### PR TITLE
Adding Golang 118

### PR DESCRIPTION
**Pull request by Toasterson**

### CODE REVIEW

**Comment by AndWac** [comment]
There are two commits in your PR; one of them doesn't belong here.

**Comment by Toasterson** [comment]
If you can tell me how to get rid of it I am all ears

File: components/developer/golang-118/golang.p5m [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #
 
 #
-# Copyright 2021 Till Wegmueller
+# Copyright 2022 Till Wegmueller
 #
 
 set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
```
Comment (AndWac) [inline components/developer/golang-118/golang.p5m:16]: This still seems odd. Adding a new package should only have added lines and no changes or deletions. This looks like git considered this have has been moved from golang-117.

**Comment by AndWac** [comment]
This doesn't build at the moment:

Building packages and commands for host, solaris/amd64.
/export2/home/andreas/oi-userland/components/developer/golang-118/build/amd64/src/crypto/elliptic/p256_asm.go:24:12: pattern p256_asm_table.bin: cannot embed irregular file p256_asm_table.bin
/export2/home/andreas/oi-userland/components/developer/golang-118/build/amd64/src/crypto/elliptic/p256_asm.go:24:12: pattern p256_asm_table.bin: cannot embed irregular file p256_asm_table.bin
go tool dist: FAILED: /export2/home/andreas/oi-userland/components/developer/golang-118/build/amd64/pkg/tool/solaris_amd64/go_bootstrap install -gcflags=all= -ldflags=all= std cmd: exit status 1
gmake: *** [/export2/home/andreas/oi-userland/make-rules/justmake.mk:62: /export2/home/andreas/oi-userland/components/developer/golang-118/build/amd64/.built] Error 2

File: components/developer/golang-118/golang.p5m (hunk 1 above)
Comment (Toasterson) [reply components/developer/golang-118/golang.p5m:16]: Well it has been.

**Comment by Toasterson** [comment]
Requires #8509 to build

**Comment by 3eka** [comment]
golang 1.18 looks like broken:
```
/usr/lib/golang/1.18/src/crypto/elliptic/p256_asm.go:24:12: pattern p256_asm_table.bin: no matching files found                                               
Command failed:     go build ./cmd/restic 
```
Compiling restic works with golang 1.17

Regards

**Pull request merged**